test/validation/api/atomic/atomic.c

```diff
@@ -369,6 +379,44 @@ static void test_atomic_cas_dec_64(void)
 	}
 }
 
+static void test_atomic_cas_inc_128(void)
+{
+	int i, ret;
+	odp_u128_t old, new;
+	odp_atomic_u128_t *a128u = &global_mem->a128u;
+
+	old = a128u->v;
+
+	for (i = 0; i < CNT; i++) {
+		do {
+			new.u64[0] = old.u64[0] + 1;
+			new.u64[1] = old.u64[1] + 1;
+
+			ret = odp_atomic_cas_u128(a128u, &old, new);
+
+		} while (ret == 0);
+	}
+}
+
+static void test_atomic_cas_dec_128(void)
+{
+	int i, ret;
+	odp_u128_t old, new;
+	odp_atomic_u128_t *a128u = &global_mem->a128u;
+
+	old = a128u->v;
```

Comment:
Please add `odp_barrier_wait(&global_mem->global_barrier);` here.

Also, now when we have `odp_atomic_load_u128()` the test can follow the same model as `test_atomic_cas_inc_32()` and `test_atomic_cas_inc_64`.